pkg/devfile/parser/configurables.go

```diff
@@ -0,0 +1,253 @@
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) removePorts() error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			component.Container.Endpoints = []common.Endpoint{}
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) hasPorts() bool {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			if len(component.Container.Endpoints) > 0 {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func (d DevfileObj) setMemory(memory string) error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			component.Container.MemoryLimit = memory
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+func (d DevfileObj) getMemory() string {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			if component.Container.MemoryLimit != "" {
+				return component.Container.MemoryLimit
+			}
+		}
+
+	}
+	return ""
+}
+
+func (d DevfileObj) getMetadataName() string {
+	return d.Data.GetMetadata().Name
+}
+
+// FormatDevfileSupportedParameters outputs supported parameters and their description
+func FormatDevfileSupportedParameters() (result string) {
+	for _, v := range GetDevfileSupportedParameters() {
+		result = result + " " + v + " - " + supportedDevfileParameterDescriptions[v] + "\n"
+	}
+	return "\nAvailable Parameters for Devfile Components:\n" + result
+}
+
+// AsDevfileSupportedParameter returns the parameter in lower case and a boolean indicating if it is a supported parameter
+func AsDevfileSupportedParameter(param string) (string, bool) {
+	lower := strings.ToLower(param)
+	return lower, lowerCaseDevfileParameters[lower]
+}
+
+// GetDevfileSupportedParameters returns the name of the supported global parameters
+func GetDevfileSupportedParameters() []string {
+	return util.GetSortedKeys(supportedDevfileParameterDescriptions)
+}
+
+func portsToEndpoints(ports ...string) ([]common.Endpoint, error) {
+	var endpoints []common.Endpoint
+	conPorts, err := util.GetContainerPortsFromStrings(ports)
+	if err != nil {
+		return nil, err
+	}
+	for _, port := range conPorts {
+
+		endpoint := common.Endpoint{
+			Name:       fmt.Sprintf("port-%d-%s", port.ContainerPort, strings.ToLower(string(port.Protocol))),
+			TargetPort: port.ContainerPort,
+			Protocol:   strings.ToLower(string(port.Protocol)),
+		}
+		endpoints = append(endpoints, endpoint)
+	}
+	return endpoints, nil
+
+}
+
+func addEndpoints(current []common.Endpoint, other []common.Endpoint) []common.Endpoint {
+	newList := make([]common.Endpoint, len(current))
+	copy(newList, current)
+	for _, ep := range other {
+		present := false
+
+		for _, presentep := range newList {
+
+			protocol := presentep.Protocol
+			if protocol == "" {
+				protocol = "tcp"
```

Comment:
The default value of endpoint protocol is `http` as defined in devfile 2.0 schema
https://devfile.github.io/devfile/_attachments/api-reference.html